src/SFA.DAS.LevyTransferMatching.Api.UnitTests/Controllers/OpportunityTests/ApplyTests.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using AutoFixture;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.LevyTransferMatching.Api.Controllers;
+using SFA.DAS.LevyTransferMatching.Api.Models.Opportunities;
+using SFA.DAS.LevyTransferMatching.Application.Commands.CreateApplication;
+
+namespace SFA.DAS.LevyTransferMatching.Api.UnitTests.Controllers.OpportunityTests
+{
+    [TestFixture]
+    public class ApplyTests
+    {
+        private OpportunityController _controller;
+        private Mock<IMediator> _mediator;
+        private readonly Fixture _fixture = new Fixture();
+
+        private int _opportunityId;
+        private long _accountId;
+        private ApplyRequest _request;
+        private CreateApplicationCommandResult _result;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _opportunityId = _fixture.Create<int>();
+            _accountId = _fixture.Create<long>();
+            _request = _fixture.Create<ApplyRequest>();
+            _result = _fixture.Create<CreateApplicationCommandResult>();
+
+            _mediator = new Mock<IMediator>();
+
+            _controller = new OpportunityController(_mediator.Object);
+
+            _mediator.Setup(x => x.Send(It.Is<CreateApplicationCommand>(command =>
+                    command.PledgeId == _opportunityId &&
```

Comment:
I don't think this is so bad. In the outer api we can act on the knowledge that an opportunity is merely a pledge and that they have the same Ids. APIM is translating the request to apply to an opportunity into the inner api's pledge-only model.

(It's also leaked a little into the UI, but that's only because we encode/decode opportunity Ids with PledgeId configuration).